libtransmission/peer-mgr.cc

```diff
@@ -1647,7 +1647,8 @@ uint64_t tr_peerMgrGetDesiredAvailable(tr_torrent const* tor)
         return 0;
     }
 
-    auto available = swarm->peers.front()->has();
+    auto available = tr_bitfield(0);
+    available.set_has_none();
```

Comment:
```suggestion
    auto available = tr_bitfield{ tor->piece_count() };
```

Reply:
I prefer my solution.
1. It avoids extra allocation.
2. On the first call to |=, it will return early:
https://github.com/transmission/transmission/blob/173c19eb5a09f2706becda72a321ad1082f987c6/libtransmission/bitfield.cc#L445-L449

Reply:
My code does not allocate and will return early.

Reply:
ah correct